.github/workflows/build-and-push-components-release.yml

```diff
@@ -0,0 +1,112 @@
+name: Build and Push Components Release
+
+on:
+  workflow_call:
+    inputs:
+      image_tag:
+        required: true
+        type: string
+        description: 'Image tag to use.'
+      registry_name:
+        required: false
+        type: string
+        description: 'Name of the registry.'
+        default: ghcr.io/openclarity
+      bake-group:
+        required: false
+        type: string
+        description: 'Name of the Docker Bake group of targets'
+        default: default
+
+jobs:
+  prepare-build:
+    runs-on: ubuntu-latest
+    outputs:
+      timestamp: ${{ steps.timestamp.outputs.timestamp }}
+      targets: ${{ steps.targets.outputs.targets }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Set timestamp
+        id: timestamp
+        run: |
+          ##
+          ## Set timestamp variable
+          ##
+
+          echo "timestamp=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> "$GITHUB_OUTPUT"
+
+      - name: List targets
+        id: targets
+        uses: docker/bake-action/subaction/list-targets@7a5dfed3550ca014665af2a27af8fc9d7284b9b3 # v4.5.0
+        with:
+          target: ${{ inputs.bake-group }}
+
+  build-and-push:
+    needs: prepare-build
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        targets: ${{ fromJson(needs.prepare-build.outputs.targets) }}
+        platform:
+          - linux/amd64
```

Comment:
In this current form, the images would be built parallel by platform in a matrix and then pushed separately instead of merging the two and then pushing one multi-arch image as in the original flow. 

We already spoke about this with Bence and concluded that 
- we don't even need to build for two platforms in case of producing the image for the e2e test, just `linux/amd64` and Docker Bake can handle this
- in case of releasing, since it happens rarely, we don't need to spare these minutes by building for the two Linux platforms parallel in a matrix and then combining those to a single multi-arch image to be pushed to ghcr.io, but we can leave this also to Docker Bake by setting `*.platform=linux/arm64,linux/amd64` in the CI step

Reply:
I have added the changes we have discussed:

- Added a workflow that can be run on any pushed branch, and it also pushes images to `ghcr.io/openclarity` with `-dev` suffix and `latest` tag:
<img width="750" alt="image" src="https://github.com/openclarity/vmclarity/assets/113284287/a7c07e98-642d-442a-91d9-6d18c6c18955">

- When a branch is merged into main, images are uploaded to `ghcr` with `latest` tag:
<img width="751" alt="image" src="https://github.com/openclarity/vmclarity/assets/113284287/e80d2a64-de9f-4e17-8bfa-22a1e2aad96c">

- When there is a release, images are pushed to `ghcr` tagged with the `release-tag`
<img width="745" alt="image" src="https://github.com/openclarity/vmclarity/assets/113284287/d8db6773-37d8-4cb2-86a7-2053bc5d5435">